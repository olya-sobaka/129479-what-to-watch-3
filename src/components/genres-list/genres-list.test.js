import React from "react";
import renderer from "react-test-renderer";
import GenresList from "./genres-list";

const movies = [
  {
    title: `Kiki's Delivery Service`,
    posterSrc: `img/midnight-special.jpg`,
    id: 36,
    genre: `horror`
  },
  {
    title: `Kiki's Delivery Service`,
    posterSrc: `img/midnight-special.jpg`,
    id: 37,
    genre: `drama`
  },
  {
    title: `Kiki's Delivery Service`,
    posterSrc: `img/midnight-special.jpg`,
    id: 38,
    genre: `documentary`
  }
];

const handleGenreClick = () => {};

it(`renders correctly`, () => {

  const activeGenre = `horror`;

  const tree = renderer.create(
      <GenresList films={movies} activeGenre={activeGenre} onGenreClick={handleGenreClick} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
