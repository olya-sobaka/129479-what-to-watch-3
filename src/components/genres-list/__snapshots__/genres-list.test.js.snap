// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ul
  className="catalog__genres-list"
>
  <li
    className="catalog__genres-item"
    onClick={[Function]}
  >
    <a
      className="catalog__genres-link"
      href="#"
    >
      All genres
    </a>
  </li>
  <li
    className="catalog__genres-item catalog__genres-item--active"
    onClick={[Function]}
  >
    <a
      className="catalog__genres-link"
      href="#"
    >
      Horror
    </a>
  </li>
  <li
    className="catalog__genres-item"
    onClick={[Function]}
  >
    <a
      className="catalog__genres-link"
      href="#"
    >
      Drama
    </a>
  </li>
  <li
    className="catalog__genres-item"
    onClick={[Function]}
  >
    <a
      className="catalog__genres-link"
      href="#"
    >
      Documentary
    </a>
  </li>
</ul>
`;
